xapian-core/queryparser/termgenerator_internal.cc

```diff
@@ -229,9 +270,23 @@ parse_terms(Utf8Iterator itor, unsigned cjk_flags, bool with_positions,
 	    do {
 		Unicode::append_utf8(term, ch);
 		prevch = ch;
-		if (++itor == Utf8Iterator() ||
-		    (cjk_flags && CJK::codepoint_is_cjk(*itor)))
+		if (++itor == Utf8Iterator())
 		    goto endofterm;
+		else if (cjk_flags) {
+		    // Only deal mixed Chinese numbers which start
+		    // with an Arabic digit and next is a Chinese digit.
+		    if (is_digit(prevch) && is_Chinese_digit(*itor)) {
```

Comment:
`is_digit()` checks for anything which is in Unicode category `DECIMAL_DIGIT_NUMBER`, which includes Chinese digits, etc - you can see the full list with:

```
grep ';Nd;' xapian-core/unicode/UnicodeData.txt
```

If you only want the ASCII digits, then the appropriate test is `C_isdigit()` (which works like the standard `isdigit()` function, except it is hardwired to the POSIX C locale, whereas `isdigit()` may match non-ASCII digit characters if the program calls `setlocale()`).  However, that takes an `unsigned char` so you need to check the value is in range for that, like how `U_isupper()` is defined:

```
static inline bool  
U_isupper(unsigned ch)
{
    return (ch < 128 && C_isupper(static_cast<unsigned char>(ch)));
}
```

If you are meaning to accept any digit character, then the comment above needs adjusting to reflect that.

Reply:
Sorry for I misunderstood the function `is_digit()`,
what I mean is to accept only ASCII digits.
I will fix it following your suggestion.